front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx

```diff
@@ -114,44 +129,91 @@ const RollingStockEditorSelect = ({
     defaultValues: {
-      startup_time: rollingStock?.startup_time || 0,
-      startup_acceleration: rollingStock?.startup_acceleration || 0,
-      comfort_acceleration: rollingStock?.comfort_acceleration || 0,
-      gamma_value: rollingStock?.gamma.value || 0,
-      inertia_coefficient: rollingStock?.inertia_coefficient || 0,
-      loading_gauge: (rollingStock?.loading_gauge as string) || '...',
-      rolling_resistance_A: rollingStock?.rolling_resistance.A || 0,
-      rolling_resistance_B: rollingStock?.rolling_resistance.B || 0,
-      rolling_resistance_C: rollingStock?.rolling_resistance.C || 0,
+      detail: rollingStockData?.metadata.detail || '',
+      family: rollingStockData?.metadata.family || '',
+      grouping: rollingStockData?.metadata.grouping || '',
+      number: rollingStockData?.metadata.number || '',
+      reference: rollingStockData?.metadata.reference || '',
+      series: rollingStockData?.metadata.series || '',
+      subseries: rollingStockData?.metadata.subseries || '',
+      type: rollingStockData?.metadata.type || '',
+      unit: rollingStockData?.metadata.unit || '',
+      length: rollingStockData?.length || 0,
+      mass: rollingStockData?.mass || 0,
+      max_speed: rollingStockData?.max_speed || 0,
+      startup_time: rollingStockData?.startup_time || 0,
+      startup_acceleration: rollingStockData?.startup_acceleration || 0,
+      comfort_acceleration: rollingStockData?.comfort_acceleration || 0,
+      gamma_value: rollingStockData?.gamma.value || 0,
+      inertia_coefficient: rollingStockData?.inertia_coefficient || 0,
+      loading_gauge: (rollingStockData?.loading_gauge as string) || 'G1',
+      rolling_resistance_A: rollingStockData?.rolling_resistance.A || 0,
+      rolling_resistance_B: rollingStockData?.rolling_resistance.B || 0,
+      rolling_resistance_C: rollingStockData?.rolling_resistance.C || 0,
+      electrical_power_startup_time: rollingStockData?.electrical_power_startup_time || 0,
+      raise_pantograph_time: rollingStockData?.raise_pantograph_time || 0,
     },
   } as UseFormProps<RollingStockParametersValues>);
 
-  // in the left side of the form, we display all the information except the rollingStock resistance params,
+  // in the top of the form we display all the metadatas
+  // the array is split in two to display the metadatas in two columns
+  const metadataForm = (position: boolean, registerForm: UseFormRegister<FieldValues>) =>
+    Object.keys(schema.properties)
+      .filter(
+        (property, index) =>
+          property.match(/detail|family|grouping|number|reference|series|subseries|type|unit/) &&
+          (position ? index < 5 : index >= 5)
+      )
+      .map((property) => (
+        <div key={property}>
+          <RollingStockEditorInput
+            label={property}
+            register={registerForm}
+            position={position}
+            error={errors[property] as unknown as string}
```

Comment:
We are not superfan of this typecasting

Reply:
just errors[property] as string should work

Reply:
ok

Reply:
done